packages/core/src/components/tabs/components/tab.tsx

```diff
@@ -0,0 +1,104 @@
+import { Component, Element, Event, EventEmitter, Prop, h, Host, Method } from "@stencil/core";
+import { isModifiedEvent } from "../../../utils/is-modified-event";
+import { TabsSwitchEvent } from "../tabs.interfaces";
+
+/**
+ * @slot - The label for this tab
+ */
+@Component({
+  tag: "dso-tab",
+  styleUrl: "tab.scss",
+  shadow: true,
+})
+export class Tab {
+  @Element()
+  host!: HTMLDsoTabElement;
+
+  /**
+   * Makes the tab active. The tab for which the tabpanel is visible is the active tab.
+   */
+  @Prop({ reflect: true })
+  active?: boolean;
+
+  /**
+   * Disables the tab. A disabled tab cannot be activated and it's tabpanel cannot be shown.
+   */
+  @Prop({ reflect: true })
+  disabled?: boolean;
+
+  /**
+   * The optional href of the tab. Creates an anchor if present. Creates a button if absent.
+   */
+  @Prop()
+  href?: string;
+
+  /**
+   * Emitted when the user activates tab via click or arrow keys followed by space or enter.
+   */
+  @Event()
+  dsoTabSwitch!: EventEmitter<TabsSwitchEvent>;
+
+  /**
+   * @internal
+   */
+  @Method()
+  async _dsoFocus() {
+    this.anchorOrButtonRef?.focus();
+  }
+
+  private anchorOrButtonRef?: HTMLAnchorElement | HTMLButtonElement;
+
+  private emitEvent = (e: MouseEvent | KeyboardEvent) => {
+    this.dsoTabSwitch.emit({
+      originalEvent: e,
+      isModifiedEvent: e instanceof MouseEvent ? isModifiedEvent(e) : false,
+    });
+  };
+
+  private keyUpHandler = (e: KeyboardEvent) => {
```

Comment:
Deze contraptie heb ik bedacht om middels Space een niet geactiveerde en niet gedisablede tab te kunnen bedienen. Bij het schrijven van de e2e echter lees ik in een [antwoord op een vraag op stackoverflow](https://stackoverflow.com/questions/66309639/clicking-a-element-using-enter-space-key-is-not-working-even-though-it-is-focu): "An anchor with a valid href tells a screen reader user "Clicking this will result in navigation". It also lets them know that they need to press Enter to activate the link." 

Dus als een dso-tab een property href heeft dan renderen wij die als een `a` en is slecht de `Enter`-key relevant voor het activeren van die link. Nu gebruik ik de keyUpHandler alleen voor het afhandelen van de Space bij een `a`.

Bij een `button` wordt de keyboard afhandeling via de clickhandler geregeld.

Dus ik denk dat die keyHandler net zo goed helemaal weg kan.

...